--- lighttpd-1.4.32.orig/src/base.h	2011-12-01 04:43:35.000000000 +0900
+++ lighttpd-1.4.32.new/src/base.h	2012-12-18 14:47:19.000000000 +0900
@@ -336,7 +336,8 @@ typedef enum {
 	CON_STATE_WRITE,
 	CON_STATE_RESPONSE_END,
 	CON_STATE_ERROR,
-	CON_STATE_CLOSE
+	CON_STATE_CLOSE,
+        CON_STATE_READ_CONTINUOUS
 } connection_state_t;
 
 typedef enum { COND_RESULT_UNSET, COND_RESULT_FALSE, COND_RESULT_TRUE } cond_result_t;
