--- lighttpd-1.4.32.orig/src/connections-glue.c	2012-11-07 23:10:18.000000000 +0900
+++ lighttpd-1.4.32.new/src/connections-glue.c	2012-12-18 14:47:19.000000000 +0900
@@ -14,6 +14,7 @@ const char *connection_get_state(connect
 	case CON_STATE_REQUEST_END: return "req-end";
 	case CON_STATE_RESPONSE_START: return "resp-start";
 	case CON_STATE_RESPONSE_END: return "resp-end";
+	case CON_STATE_READ_CONTINUOUS: return "read-continuous";
 	default: return "(unknown)";
 	}
 }
@@ -31,6 +32,7 @@ const char *connection_get_short_state(c
 	case CON_STATE_REQUEST_END: return "Q";
 	case CON_STATE_RESPONSE_START: return "s";
 	case CON_STATE_RESPONSE_END: return "S";
+	case CON_STATE_READ_CONTINUOUS: return "F";
 	default: return "x";
 	}
 }
