--- lighttpd-1.4.29.orig/src/connections-glue.c	2012-01-18 11:45:10.684414394 +0900
+++ lighttpd-1.4.29.new/src/connections-glue.c	2012-01-24 13:07:26.039565849 +0900
@@ -14,6 +14,7 @@
 	case CON_STATE_REQUEST_END: return "req-end";
 	case CON_STATE_RESPONSE_START: return "resp-start";
 	case CON_STATE_RESPONSE_END: return "resp-end";
+	case CON_STATE_READ_CONTINUOUS: return "read-continuous";
 	default: return "(unknown)";
 	}
 }
@@ -31,6 +32,7 @@
 	case CON_STATE_REQUEST_END: return "Q";
 	case CON_STATE_RESPONSE_START: return "s";
 	case CON_STATE_RESPONSE_END: return "S";
+	case CON_STATE_READ_CONTINUOUS: return "F";
 	default: return "x";
 	}
 }
