--- lighttpd-1.4.31.orig/src/Makefile.am	2012-06-20 09:35:22.000000000 +0900
+++ lighttpd-1.4.31.new/src/Makefile.am	2012-06-20 10:09:19.000000000 +0900
@@ -265,6 +265,24 @@
 mod_accesslog_la_LDFLAGS = -module -export-dynamic -avoid-version -no-undefined
 mod_accesslog_la_LIBADD = $(common_libadd)
 
+if WITH_WEBSOCKET
+lib_LTLIBRARIES += mod_websocket.la
+
+mod_websocket_la_SOURCES = mod_websocket.h mod_websocket.c base64.c mod_websocket_connector.c mod_websocket_handshake.c mod_websocket_frame.c
+
+if WITH_LIBICU
+mod_websocket_la_SOURCES += mod_websocket_conv.c
+endif
+
+if !WEBSOCKET_USE_OPENSSL
+mod_websocket_la_SOURCES += sha1.h sha1.c
+endif
+
+mod_websocket_la_CFLAGS = $(AM_CFLAGS) $(WEBSOCKET_VERSION_CFLAGS) $(WEBSOCKET_ICU_CFLAGS)
+mod_websocket_la_LDFLAGS = -module -export-dynamic -avoid-version -no-undefined
+mod_websocket_la_LIBADD = $(common_libadd) $(WEBSOCKET_ICU_LIBS)
+
+endif
 
 hdr = server.h buffer.h network.h log.h keyvalue.h \
       response.h request.h fastcgi.h chunk.h \
