diff -ur lighttpd-1.4.28.orig/configure.ac lighttpd-1.4.28/configure.ac
--- lighttpd-1.4.28.orig/configure.ac	2010-08-13 20:10:15.000000000 +0900
+++ lighttpd-1.4.28/configure.ac	2011-06-20 08:48:53.000000000 +0900
@@ -81,7 +81,7 @@
 AC_TYPE_SIGNAL
 AC_FUNC_STAT
 AC_FUNC_STRFTIME
-AC_CHECK_FUNCS([issetugid inet_pton])
+AC_CHECK_FUNCS([issetugid inet_pton strtoul nl_langinfo setlocale])
 
 if test -z "$PKG_CONFIG"; then
   AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
@@ -215,6 +215,73 @@
  AC_SUBST(ATTR_LIB)
 fi
 
+dnl Check for iconv
+AC_MSG_CHECKING(whether --with-libiconv option specified)
+AC_ARG_WITH(
+        libiconv,
+        [  --with-libiconv=DIR  enable to convert locale: default=yes],
+        [ libiconv_dir=$withval ],
+        [ libiconv_dir=yes ])
+
+if test "x${libiconv_dir}" != "xno"; then
+   AC_MSG_RESULT($libiconv_dir)
+   LIBICONV_CFLAGS="-I/usr/local/include"
+   LIBICONV_LDFLAGS="-L/usr/local/lib"
+   if test "x${libiconv_dir}" != "xyes"; then
+      LIBICONV_CFLAGS="-I${libiconv_dir}/include"
+      LIBICONV_LDFLAGS="-L${libiconv_dir}/lib"
+   fi
+   save_CFLAGS="$CFLAGS"
+   save_LDFLAGS="$LDFLAGS"
+   CFLAGS=$LIBICONV_CFLAGS
+   LDFLAGS=$LIBICONV_LDFLAGS
+   AC_CHECK_FUNC(iconv, has_iconv=yes, has_iconv=no)
+   AC_CHECK_LIB(iconv, iconv, has_libiconv=yes, has_libiconv=no)
+   if test "x${has_iconv}" == "xno" &&
+      test "x${has_libiconv}" == "xno"; then
+      HAS_LIBICONV=""
+   else
+      HAS_LIBICONV="yes"
+   fi
+   if test "x${has_iconv}" == "xno" &&
+      test "x${has_libiconv}" == "xyes"; then
+           LIBICONV_LIBS="-liconv"
+   fi
+   CFLAGS="$save_CFLAGS"
+   LDFLAGS="$save_LDFLAGS"
+fi
+AC_SUBST(HAS_LIBICONV)
+AC_SUBST(LIBICONV_CFLAGS)
+AC_SUBST(LIBICONV_LDFLAGS)
+AC_SUBST(LIBICONV_LIBS)
+CFLAGS="$CFLAGS $LIBICONV_CFLAGS"
+LDFLAGS="$LDFLAGS $LIBICONV_LDFLAGS"
+LIBS="$LIBS $LIBICONV_LIBS"
+
+dnl Check for websocket version
+AC_MSG_CHECKING(whether --with-websocket option specified)
+AC_ARG_WITH(
+        websocket,
+        [  --with-websocket@<:@=IETF-{00,08}@:>@
+                                  specify draft version : default=IETF-00],
+        [ websocket=$withval ],
+        [ websocket=no ])
+AC_MSG_RESULT(${websocket})
+if test "x${websocket}" == "xyes" ||
+   test "x${websocket}" == "xietf-00" ||
+   test "x${websocket}" == "xIETF-00"; then
+   websocket="IETF-00"
+   WEBSOCKET_VERSION_CFLAGS="-D_MOD_WEBSOCKET_SPEC_IETF_00_"
+elif test "x${websocket}" == "xietf-08" ||
+     test "x${websocket}" == "xIETF-08"; then
+   WEBSOCKET_VERSION_CFLAGS="-D_MOD_WEBSOCKET_SPEC_IETF_08_"
+elif test "x${websocket}" != "xno"; then
+   AC_MSG_ERROR([websocket option is invalid. plz specify IETF-{00,08}])
+fi
+AC_SUBST(websocket)
+AC_SUBST(WEBSOCKET_VERSION_CFLAGS)
+CFLAGS="$CFLAGS $WEBSOCKET_VERSION_CFLAGS"
+
 dnl openssl on solaris needs -lsocket -lnsl
 AC_SEARCH_LIBS(socket,socket)
 AC_SEARCH_LIBS(gethostbyname,nsl socket)
@@ -625,6 +692,17 @@
 do_build="mod_cgi mod_fastcgi mod_extforward mod_proxy mod_evhost mod_simple_vhost mod_access mod_alias mod_setenv mod_usertrack mod_auth mod_status mod_accesslog"
 do_build="$do_build mod_rrdtool mod_secdownload mod_expire mod_compress mod_dirlisting mod_indexfile mod_userdir mod_webdav mod_staticfile mod_scgi mod_flv_streaming"
 
+if test "x${websocket}" != "xno"; then
+        plugins="mod_websocket(${websocket})"
+        if test ! "x$HAS_LIBICONV" = x; then
+                do_build="$do_build $plugins"
+        else
+                no_build="$no_build $plugins"
+        fi
+else
+        no_build="$no_build mod_websocket"
+fi
+
 plugins="mod_rewrite mod_redirect mod_ssi mod_trigger_b4_dl"
 features="regex-conditionals"
 if test ! "x$PCRE_LIB" = x; then
lighttpd-1.4.28だけに発見: configure.ac~
diff -ur lighttpd-1.4.28.orig/src/Makefile.am lighttpd-1.4.28/src/Makefile.am
--- lighttpd-1.4.28.orig/src/Makefile.am	2010-08-12 05:44:17.000000000 +0900
+++ lighttpd-1.4.28/src/Makefile.am	2011-06-20 08:47:26.000000000 +0900
@@ -95,6 +95,11 @@
 common_libadd =
 endif
 
+lib_LTLIBRARIES += mod_websocket.la
+mod_websocket_la_SOURCES = mod_websocket.c sha1.c
+mod_websocket_la_LDFLAGS = -module -export-dynamic -avoid-version -no-undefined
+mod_websocket_la_LIBADD = $(common_libadd)
+
 lib_LTLIBRARIES += mod_flv_streaming.la
 mod_flv_streaming_la_SOURCES = mod_flv_streaming.c
 mod_flv_streaming_la_LDFLAGS = -module -export-dynamic -avoid-version -no-undefined
diff -ur lighttpd-1.4.28.orig/src/base.h lighttpd-1.4.28/src/base.h
--- lighttpd-1.4.28.orig/src/base.h	2010-08-07 21:44:17.000000000 +0900
+++ lighttpd-1.4.28/src/base.h	2011-06-20 08:47:26.000000000 +0900
@@ -331,7 +331,8 @@
 	CON_STATE_WRITE,
 	CON_STATE_RESPONSE_END,
 	CON_STATE_ERROR,
-	CON_STATE_CLOSE
+	CON_STATE_CLOSE,
+	CON_STATE_READ_CONTINUOUS,
 } connection_state_t;
 
 typedef enum { COND_RESULT_UNSET, COND_RESULT_FALSE, COND_RESULT_TRUE } cond_result_t;
diff -ur lighttpd-1.4.28.orig/src/connections-glue.c lighttpd-1.4.28/src/connections-glue.c
--- lighttpd-1.4.28.orig/src/connections-glue.c	2010-08-17 18:04:38.000000000 +0900
+++ lighttpd-1.4.28/src/connections-glue.c	2011-06-20 08:47:26.000000000 +0900
@@ -14,6 +14,7 @@
 	case CON_STATE_REQUEST_END: return "req-end";
 	case CON_STATE_RESPONSE_START: return "resp-start";
 	case CON_STATE_RESPONSE_END: return "resp-end";
+	case CON_STATE_READ_CONTINUOUS: return "read-continuous";
 	default: return "(unknown)";
 	}
 }
@@ -31,6 +32,7 @@
 	case CON_STATE_REQUEST_END: return "Q";
 	case CON_STATE_RESPONSE_START: return "s";
 	case CON_STATE_RESPONSE_END: return "S";
+	case CON_STATE_READ_CONTINUOUS: return "F";
 	default: return "x";
 	}
 }
diff -ur lighttpd-1.4.28.orig/src/connections.c lighttpd-1.4.28/src/connections.c
--- lighttpd-1.4.28.orig/src/connections.c	2010-08-17 18:04:38.000000000 +0900
+++ lighttpd-1.4.28/src/connections.c	2011-06-20 08:47:26.000000000 +0900
@@ -1163,6 +1163,8 @@
 		}
 
 		break;
+	case CON_STATE_READ_CONTINUOUS:
+		break;
 	default: break;
 	}
 
@@ -1236,7 +1238,8 @@
 	}
 
 	if (con->state == CON_STATE_READ ||
-	    con->state == CON_STATE_READ_POST) {
+	    con->state == CON_STATE_READ_POST ||
+		con->state == CON_STATE_READ_CONTINUOUS) {
 		connection_handle_read_state(srv, con);
 	}
 
@@ -1638,12 +1641,18 @@
 			break;
 		case CON_STATE_READ_POST:
 		case CON_STATE_READ:
+		case CON_STATE_READ_CONTINUOUS:
 			if (srv->srvconf.log_state_handling) {
 				log_error_write(srv, __FILE__, __LINE__, "sds",
 						"state for fd", con->fd, connection_get_state(con->state));
 			}
 
 			connection_handle_read_state(srv, con);
+
+            if (con->state == CON_STATE_READ_CONTINUOUS) {
+                plugins_call_read_continuous(srv, con);
+            }
+
 			break;
 		case CON_STATE_WRITE:
 			if (srv->srvconf.log_state_handling) {
@@ -1807,6 +1816,9 @@
 			fdevent_event_del(srv->ev, &(con->fde_ndx), con->fd);
 		}
 		break;
+	case CON_STATE_READ_CONTINUOUS:
+		/* leave up to plugins */
+		break;
 	default:
 		fdevent_event_del(srv->ev, &(con->fde_ndx), con->fd);
 		break;
diff -ur lighttpd-1.4.28.orig/src/plugin.c lighttpd-1.4.28/src/plugin.c
--- lighttpd-1.4.28.orig/src/plugin.c	2010-08-17 18:04:38.000000000 +0900
+++ lighttpd-1.4.28/src/plugin.c	2011-06-20 08:47:26.000000000 +0900
@@ -44,6 +44,7 @@
 		PLUGIN_FUNC_INIT,
 		PLUGIN_FUNC_CLEANUP,
 		PLUGIN_FUNC_SET_DEFAULTS,
+		PLUGIN_FUNC_READ_CONTINUOUS,
 
 		PLUGIN_FUNC_SIZEOF
 } plugin_t;
@@ -274,6 +275,7 @@
 PLUGIN_TO_SLOT(PLUGIN_FUNC_HANDLE_DOCROOT, handle_docroot)
 PLUGIN_TO_SLOT(PLUGIN_FUNC_HANDLE_PHYSICAL, handle_physical)
 PLUGIN_TO_SLOT(PLUGIN_FUNC_CONNECTION_RESET, connection_reset)
+PLUGIN_TO_SLOT(PLUGIN_FUNC_READ_CONTINUOUS, read_continuous)
 
 #undef PLUGIN_TO_SLOT
 
@@ -403,6 +405,7 @@
 		PLUGIN_TO_SLOT(PLUGIN_FUNC_HANDLE_DOCROOT, handle_docroot);
 		PLUGIN_TO_SLOT(PLUGIN_FUNC_HANDLE_PHYSICAL, handle_physical);
 		PLUGIN_TO_SLOT(PLUGIN_FUNC_CONNECTION_RESET, connection_reset);
+		PLUGIN_TO_SLOT(PLUGIN_FUNC_READ_CONTINUOUS, read_continuous);
 		PLUGIN_TO_SLOT(PLUGIN_FUNC_CLEANUP, cleanup);
 		PLUGIN_TO_SLOT(PLUGIN_FUNC_SET_DEFAULTS, set_defaults);
 #undef PLUGIN_TO_SLOT
diff -ur lighttpd-1.4.28.orig/src/plugin.h lighttpd-1.4.28/src/plugin.h
--- lighttpd-1.4.28.orig/src/plugin.h	2010-02-02 08:28:20.000000000 +0900
+++ lighttpd-1.4.28/src/plugin.h	2011-06-20 08:47:26.000000000 +0900
@@ -23,6 +23,7 @@
 #define PHYSICALPATH_FUNC  CONNECTION_FUNC
 #define REQUESTDONE_FUNC   CONNECTION_FUNC
 #define URIHANDLER_FUNC    CONNECTION_FUNC
+#define READ_CONT_FUNC     CONNECTION_FUNC
 
 #define PLUGIN_DATA        size_t id
 
@@ -55,6 +56,7 @@
 											    */
 	handler_t (* handle_subrequest)      (server *srv, connection *con, void *p_d);    /* */
 	handler_t (* connection_reset)       (server *srv, connection *con, void *p_d);    /* */
+	handler_t (* read_continuous)        (server *srv, connection *con, void *p_d);    /* */
 	void *data;
 
 	/* dlopen handle */
@@ -74,6 +76,7 @@
 handler_t plugins_call_handle_connection_close(server *srv, connection *con);
 handler_t plugins_call_handle_joblist(server *srv, connection *con);
 handler_t plugins_call_connection_reset(server *srv, connection *con);
+handler_t plugins_call_read_continuous(server *srv, connection *con);
 
 handler_t plugins_call_handle_trigger(server *srv);
 handler_t plugins_call_handle_sighup(server *srv);
